src/core/dev/ring.h

```diff
@@ -92,6 +92,21 @@ class ring {
                                   xlio_wr_tx_packet_attr attr) = 0;
     virtual int send_lwip_buffer(ring_user_id_t id, xlio_ibv_send_wr *p_send_wqe,
                                  xlio_wr_tx_packet_attr attr, xlio_tis *tis) = 0;
+    virtual uint32_t send_doca_single(void *ptr, uint32_t len, mem_buf_desc_t *buff)
```

Comment:
Do not implement those methods. Set them as abstract. Let the child classes implement them event with empty implementation.

Reply:
done.